File: lecture0-30/lecture002/starpattern.cpp
#include<iostream>

using namespace std;

int main(int args,char**argv)
{   int n;
    
    cout<<"enter no of layers : ";
    cin>>n;
    int nst=1;
    int nsp=n/2;

    for(int r=1;r<=n;r++)
    {
        for(int csp=1;csp<=nsp;csp++){
            cout<<" ";
        }
        for(int cst=1;cst<=nst;cst++){
            cout<<"*";  
        }
        if(r<=n/2){
        nst+=2;
        nsp--;
        }
        else{
        nst-=2;
        nsp++;
        }
        cout<<endl;
        
    }
}

File: lecture002/starpattern.java
import java.util.Scanner;

public class starpattern {
    public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);
        int n;

        System.out.print("enter no of layers : ");
        n = scn.nextInt();

        int nst = 1;
        int nsp = n / 2;

        for (int r = 0; r <= n; r++) {
            for (int csp = nsp; csp >= 1; csp--) {
                System.out.print(" ");
            }
            for (int cst = 1; cst <= nst; cst++) {
                System.out.print("*");
            }

            if (r < n / 2) {
                nsp--;
                nst += 2;
            } else {
                nsp++;
                nst -= 2;
            }
            System.out.println();
        }
    }
}